src/OVAL/oval_generator.c

```diff
@@ -138,9 +156,30 @@ void oval_generator_set_timestamp(struct oval_generator *generator, const char *
 	generator->timestamp = oscap_strdup(timestamp);
 }
 
+void oval_generator_update_timestamp(struct oval_generator *generator)
+{
+	time_t et;
+	struct tm *lt;
+	char timestamp[] = "yyyy-mm-ddThh:mm:ss";
+
+	time(&et);
+	lt = localtime(&et);
+	snprintf(timestamp, sizeof(timestamp), "%4d-%02d-%02dT%02d:%02d:%02d",
+		 1900 + lt->tm_year, 1 + lt->tm_mon, lt->tm_mday, lt->tm_hour, lt->tm_min, lt->tm_sec);
+	oval_generator_set_timestamp(generator, timestamp);
+}
+
+void oval_generator_add_platform_schema_version(struct oval_generator *generator, const char *platform, const char *schema_version)
+{
+	oscap_htable_add(generator->platform_schema_versions, platform, oscap_strdup(schema_version));
+}
+
+
 xmlNode *oval_generator_to_dom(struct oval_generator *generator, xmlDocPtr doc, xmlNode *parent)
 {
-	xmlNode *gen_node;
+	struct oscap_htable_iterator *sv_itr;
+	const char *platform, *version;
```

Comment:
You declare `platform` up here, it is only used in the while loop. At the same time you declare `platform_uri` and `namespace_uri_length` just in the while loop.

Opinions differ on this but I prefer to bind variables to as small scopes as possible. Either way, choose one way and stick to it please.